### DISCUSSION TITLE
Error while doing HOROVOD installation

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#2192] [Q&A] [answered]
Hi,

Facing error while installing Horovod.

Command -- HOROVOD_WITHOUT_GLOO=1 HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_OPERATIONS=NCCL HOROVOD_NCCL_HOME=/path/to/nccl pip install horovod mpi4py

Output --

![Screenshot 2022-12-19 at 5 52 06 PM](https://user-images.githubusercontent.com/10074710/208430200-9333206c-37a5-4cb9-b701-e18d52e92301.png)

![Screenshot 2022-12-19 at 5 57 00 PM](https://user-images.githubusercontent.com/10074710/208430509-dee5a9f2-1417-4962-8d0c-316e9162b2ba.png)

![Screenshot 2022-12-19 at 5 57 14 PM](https://user-images.githubusercontent.com/10074710/208430540-16e5c476-d5ed-44d1-b4ac-dac8e2e780cd.png)

![Screenshot 2022-12-19 at 5 57 22 PM](https://user-images.githubusercontent.com/10074710/208430559-50e1a775-620f-478f-b3c5-e7c30902887d.png)

**Comment by njzjz** [comment, marked as answer]
Figure 1 is the error for horovod. Figures 2-4 are the error for mpi4py. I suggest asking in the upstream repository.
https://github.com/horovod/horovod
https://github.com/mpi4py/mpi4py
